RecoLocalCalo/EcalRecProducers/plugins/Common.h

```diff
@@ -0,0 +1,46 @@
+#ifndef RecoLocalCalo_EcalRecProducers_plugins_Common_h
+#define RecoLocalCalo_EcalRecProducers_plugins_Common_h
+
+#include <cassert>
+#include <chrono>
+#include <cmath>
+#include <cstdint>
+#include <iostream>
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+// a workaround for std::abs not being a constexpr function
+namespace ecal {
+
+  template <typename T>
+  constexpr T abs(T const& value) {
+    return ::std::max(value, -value);
+  }
+
+  // temporary
+  namespace mgpa {
+
+    constexpr int adc(uint16_t sample) { return sample & 0xfff; }
+    constexpr int gainId(uint16_t sample) { return (sample >> 12) & 0x3; }
```

Comment:
Can we take the values `0xfff`, `12` and `0x3` from [DataFormats/EcalDigi/interface/EcalConstants.h](https://github.com/cms-sw/cmssw/tree/master/DataFormats/EcalDigi/interface/EcalConstants.h) ?

Reply:
In fact, can we directly reuse the functions from [DataFormats/EcalDigi/interface/EcalLiteDTUSample.h](https://github.com/cms-sw/cmssw/blob/master/DataFormats/EcalDigi/interface/EcalLiteDTUSample.h) ?

I'm definitely no expert in ECAL - so I'm curious, why these constants are in the `ecalPh2` namespace (class, actually)?
Are we using for Run 2 / Run 3 something that is supposedly only for Phase 2 ?

Reply:
No I would not use the functions from EcalLiteDTUSample since this is indeed only for Phase2. My guess is that for the current system these masks are hard coded. I think moving them to the EcalConstants.h would be a good thing to do but outside of the scope of this PR. The masks are not going to change so it would be only for better readability of the code.

Reply:
> No I would not use the functions from EcalLiteDTUSample since this is indeed only for Phase2.

But - they are using the same constants and doing the same thing ?

Reply:
just fyi, the constants for these 2 functions are from `DataFormats/EcalDigi/interface/EcalMGPASample .h`. since they were hardcoded in there, i hardcoded them for gpu ones as well. i think the reason for not using the package directly is cause when i was doing things, the pr for ecal to make things constexpr was not merged at that point...

Reply:
> > No I would not use the functions from EcalLiteDTUSample since this is indeed only for Phase2.
> 
> But - they are using the same constants and doing the same thing ?

In theory the LiteDTU ones could still change since the final design is not yet fixed. The fact that the MGPA and LiteDTU use the same masks is a coincidence.

Reply:
Discussion moved to https://github.com/cms-sw/cmssw/issues/32480#issuecomment-744578708